src/hypothesis/strategies.py

```diff
@@ -412,13 +413,17 @@ def sampled_from(elements):
 
     from hypothesis.searchstrategy.misc import SampledFromStrategy, \
         JustStrategy
-    elements = tuple(iter(elements))
+    from hypothesis.internal.conjecture.utils import check_sample
     if not elements:
         return nothing()
     if len(elements) == 1:
-        return JustStrategy(elements[0])
-    else:
-        return SampledFromStrategy(elements)
+        return JustStrategy(next(iter(elements)))
+    try:
+        elements = check_sample(elements)
+    except InvalidArgument as err:
+        elements = tuple(elements)
+        note_deprecation(err.args[0] if err.args else u'Unorderable elements')
```

Comment:
Yeah, I'll move the deprecation logic to `check_sample`; we can switch that over to an invalid argument error at the end of the deprecation period.